elastic/kibana
# [Logstash] Migrate code editor to monaco

Fixes https://github.com/elastic/kibana/issues/155842

## Summary

This PR removes the dependency it has on ace editor in favour of Monaco. Note that `CodeEditor` is a kibana wide abstraction built on top of Monaco and located at `@kbn/kibana-react-plugin/public`.

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/es-backups` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Logstash` -> `Create Pipeline`
3. Verify that the editor is Monaco

<details>
<summary>Screenshots</summary>

// Before
![Screenshot 2023-07-28 at 11 15 44](https://github.com/elastic/kibana/assets/1191206/a3cc31b7-ffb1-4caa-8dbb-5024a2348a13)

// After
![Screenshot 2023-07-28 at 11 15 39](https://github.com/elastic/kibana/assets/1191206/9e4bb832-81c1-4012-a991-ea8664d1a47a)


</details>
